### PR TITLE
Fix CM UI empty accounts from unstable connection

### PR DESCRIPTION
**Pull request by stegaBOB**
When the txns need to be split up due to size constraints, this PR handles the transaction individually and saves the state of the first transaction to prevent duplicate setup accounts.

### CODE REVIEW

File: js/packages/candy-machine-ui/src/Home.tsx [hunk 1]
```diff
@@ -51,12 +54,15 @@ const Home = (props: HomeProps) => {
...
     severity: undefined,
   });
+
   const [isActive, setIsActive] = useState(false);
   const [endDate, setEndDate] = useState<Date>();
   const [itemsRemaining, setItemsRemaining] = useState<number>();
   const [isWhitelistUser, setIsWhitelistUser] = useState(false);
   const [isPresale, setIsPresale] = useState(false);
   const [discountPrice, setDiscountPrice] = useState<anchor.BN>();
+  const [neexTxnSplit, setNeedTxnSplit] = useState(true);
```
Comment (goriunov) [inline js/packages/candy-machine-ui/src/Home.tsx:64]: Should be needTxnSplit

**Review by goriunov — CHANGES_REQUESTED** [review]

File: js/packages/candy-machine-ui/src/Home.tsx (hunk 1 above)
Comment (stegaBOB) [reply js/packages/candy-machine-ui/src/Home.tsx:64]: Thanks!

Comment (stegaBOB) [inline js/packages/candy-machine-ui/src/Home.tsx:64]:
```suggestion
  const [needTxnSplit, setNeedTxnSplit] = useState(true);
```

File: js/packages/candy-machine-ui/src/Home.tsx [hunk 2]
```diff
@@ -215,14 +236,59 @@ const Home = (props: HomeProps) => {
     try {
       setIsUserMinting(true);
       document.getElementById('#identity')?.click();
+      let setupMint: SetupState | undefined;
       if (wallet.connected && candyMachine?.program && wallet.publicKey) {
+        if (neexTxnSplit && setupTxn === undefined) {
```
Comment (goriunov) [inline js/packages/candy-machine-ui/src/Home.tsx:241]: Probably should rename the rest of the variables as well 😄 neexTxnSplit  -> needTxnSplit

**Review by goriunov — CHANGES_REQUESTED** [review]

File: js/packages/candy-machine-ui/src/Home.tsx (hunk 2 above)
Comment (stegaBOB) [reply js/packages/candy-machine-ui/src/Home.tsx]: Whoops lol

**Review by samuelvanderwaal — APPROVED** [review]

**Review by roederw — APPROVED** [review]

**Pull request merged**